Array_Problem_Task_Recursion.js

```diff
@@ -0,0 +1,48 @@
+const array2 = [
+  { id: "one", next: "two", value: null },
+  { id: "two", next: "three", value: null },
+  { id: "three", next: null, value: null },
+];
+const newArray = [];
+array2.forEach((element) => {
+  if (element.next === null) {
+    newArray.push(element);
+  } else {
+    let newobj = { ...element };
+    newArray.push(arrrayProblemFUnction(newobj));
+  }
+});
+function arrrayProblemFUnction(arrayobject) { //recursive Function
+  if (arrayobject.next !== null) {
+    let check = findingNextObject(arrayobject.next);
+    if (check === false) {
+      console.log("\n\nSorry Next object Not Found\n\n");
+      return;
+    } else {
+      arrayobject.value = check;
+    }
+    arrrayProblemFUnction(arrayobject.value);
+    return arrayobject;
+  } else {
+    return arrayobject;
```

Comment:
Never manipulate a parameter object, always create new object of it and return that.

Reply:
Noted.